source/android/adaptivecards/src/main/java/io/adaptivecards/renderer/readonly/TableCellRenderer.kt

```diff
@@ -0,0 +1,107 @@
+import io.adaptivecards.objectmodel.*
+import io.adaptivecards.renderer.*
+import io.adaptivecards.renderer.actionhandler.ICardActionHandler
+import io.adaptivecards.renderer.registration.CardRendererRegistration
+
+open class TableCellRenderArgs(renderArgs: RenderArgs, val rowIndex: Int, val colIndex: Int, val table: Table, val tableLayout: TableLayout) : RenderArgs(renderArgs)
+
+object TableCellRenderer : BaseCardElementRenderer() {
+    override fun render(renderedCard: RenderedAdaptiveCard, context: Context, fragmentManager: FragmentManager, viewGroup: ViewGroup, baseCardElement: BaseCardElement, cardActionHandler: ICardActionHandler?, hostConfig: HostConfig, renderArgs: RenderArgs): View {
+        if (renderArgs !is TableCellRenderArgs) throw IllegalArgumentException("renderArgs must be instance of TableCellRenderArgs")
+
+        val row = renderArgs.table.GetRows()[renderArgs.rowIndex]
+        val col = renderArgs.table.GetColumns()[renderArgs.colIndex]
+
+        val cell = Util.castTo(baseCardElement, TableCell::class.java)
+        val cellLayout = LinearLayout(context).apply {
+            orientation = LinearLayout.VERTICAL
+            tag = TagContent(cell)
+            layoutParams = TableRow.LayoutParams(0, TableLayout.LayoutParams.MATCH_PARENT)
+        }
+
+        cellLayout.updateLayoutParams<ViewGroup.MarginLayoutParams> {
+            val isFirst = renderArgs.colIndex == 0
+            val isLast = renderArgs.colIndex == renderArgs.table.GetColumns().size - 1
+
+            if (renderArgs.table.GetShowGridLines()) {
+                // For grid lines, draw rectangle
+                cellLayout.background = GradientDrawable().apply {
+                    shape = GradientDrawable.RECTANGLE
+                    setColor(Color.TRANSPARENT)
+
+                    // border color from config
+                    val borderColor = Color.parseColor(hostConfig.GetBorderColor(renderArgs.table.GetGridStyle()))
+                    setStroke(Util.dpToPixels(context, 1f), borderColor)
+                }
+            } else if (cell.GetBleed()) {
+                // To bleed, negate margins of first/last cells
+                val padding = Util.dpToPixels(context, hostConfig.GetSpacing().paddingSpacing.toFloat())
+                if (isFirst) marginStart = -padding
+                if (isLast) marginEnd = -padding
+            } else {
+                // Otherwise, add spacing between cells
+                val spacing = Util.dpToPixels(context, hostConfig.GetTable().cellSpacing.toFloat())
+                if (!isFirst) marginStart = spacing / 2
+                if (!isLast) marginEnd = spacing / 2
+            }
+        }
+
+        if (col.GetWidth() != null) {
+            // Weighted columns can stretch/shrink
+            cellLayout.updateLayoutParams<TableRow.LayoutParams> { width = 0; weight = col.GetWidth()!!.toFloat() }
+            renderArgs.tableLayout.setColumnShrinkable(renderArgs.colIndex, true)
```

Comment:
can this lines be taken out of the if/else blocks? looks like the assignment you want to make should be:

renderArgs.tableLayout.setColumnShrinkable(renderArgs.colIndex, col.GetWidth() != null)

but maybe I'm missing other types of column width?